_specs/ecip-51-attack-protection-system.md

```diff
@@ -0,0 +1,157 @@
+
+<< The exact format will be specified later >>
+
+### Inclusion of VisibilityProof Proofs
+
+On a first version, EtHash miners will be the only ones that will have the possibility to add VisibilityProofs to their blocks.
+
+In the future, the ETC network can propagate VisibilityProofs messages and we can let any user participate and collect fees from this system. This requires no changes to consensus. 
+
+### Rewards
+
+If a miner decides to add a VisibilityProof message in a slot, it will be rewarded by an extra subsidy. To prevent changing the ETC supply function, the extra subsidy will be obtained by diluting all remaining block subsidies by a fixed percentage. The exact amount is to be decided by the community but we propose this to be 10% of the current subsidy, starting at 1% and increasing until 10% during a period of 3 months. This will allow for EtHash miners to adapt and provides enough incentives for them to start using the RSK UMM system to obtain VisibilityProofs for their blocks.
+
+To enable the creation of an independent and open VisibilityProofs market, a small percentage (i.e. 5%) of the Visibility subsidy is paid to the miner who included the VisibilityProof message. The remainder is paid to the ETC-Address (funds are directly added to the account balance without contract code execution). If the miner address matches the ETC-Address, then all reward will be paid to the same account.
+
+
+### Timestamp Validation
+
+The timestamp of Bitcoin hit header must be not higher than 10 minutes and not lower than 10 minutes of thetimestamp of the ETC checkpoint referenced by the VisibilityProof. This prevents Bitcoin miners from creating visibility proofs for old or future branches of the ETC blockchain.
+
+### JSON-RPC Interface
+Expose two new methods:
+- *getSHAWork:* retrieves SHAsh to be used for merged mining with Bitcoin.
+- *submitSHAPoW:* receives all the needed elements to validate the PoW and then create the UMM uncle block.
+
+
+## Rationale
+
+The spirit of this proposal is to present a simple way to provide ETC with an otional system that can temporarily or permanently protect it from 51% attacks by establishing a strong deterrent. To achieve that:
+
+* VisibilityProof messages are included as uncles so the system is the least invasive way as possible.
+* In the current version, VisibilityProof messages are included by mining nodes. However, the system allows any user to participate in the VisibilityProof system.  This extension requires adding a new message to the wire protocol the relay of VisibilityProof proofs. 
+* By having a non-mandatory addition of VisibilityProof messages, ETC blockchain can keep working normally even if with no VisibilityProof on the blocks. 
+
+## Universal Merged Mining
+
+Universal Merged Mining offers access to Bitcoin SHA-256 PoW by forwarding the to-be-validated information through smart-contracts deployed on the RSK blockchain. 
```

Comment:
I understand where you are coming from, could you elaborate why RSK and what feature is missing from Classic to leverage such a functionality on Classic directly?

Reply:
RSK has direct access to ~40% (sometimes >50%) to the Bitcoin miners. Smart contracts can communicate directly with Bitcoin miners, and Bitcoin miners can collect revenue directly from these smart-contracts. By using the auction contracts in RSK the ETC community leverages on this established technical and economic channel.

Reply:
The doesn't mean that, in the future, ETC can deploy their own ETC nodes for Bitcoin mining pools. RSK is currently the most cost-efficient way to do it, and it's 100% operational.